lib/trento_web/views/error_view.ex

```diff
@@ -1,24 +1,97 @@
 defmodule TrentoWeb.ErrorView do
   use TrentoWeb, :view
 
-  # If you want to customize a particular status code
-  # for a certain format, you may uncomment below.
-  # def render("500.html", _assigns) do
-  #   "Internal Server Error"
-  # end
-
-  # By default, Phoenix returns the status message from
-  # the template name. For example, "404.html" becomes
-  # "Not Found".
   def template_not_found(template, _assigns) do
-    Phoenix.Controller.status_message_from_template(template)
+    %{
+      errors: [
+        %{
+          title: Phoenix.Controller.status_message_from_template(template),
+          detail: "An error has occurred."
+        }
+      ]
+    }
   end
 
-  def render("error.json", %{reason: reason}) do
-    %{error: reason}
+  def render("400.json", %{reason: %{exception: exception}}) do
+    %{
+      errors: [
+        %{
+          title: "Bad Request",
+          detail: Exception.message(exception)
+        }
+      ]
+    }
   end
 
-  def render("errors.json", %{errors: errors}) do
-    %{errors: errors}
+  def render("401.json", %{detail: detail}) do
+    %{
+      errors: [
+        %{
+          title: "Unauthorized",
+          detail: detail
+        }
+      ]
+    }
+  end
+
+  def render("422.json", %{error: error}) when is_map(error) do
```

Comment:
I find a bit strange to find `error` as argument some times and `detail` others, but I guess it might have some reason about how phonix and plug handle this errors

Reply:
this is going to be refactored soon as we don't want the fallback controller to act this way so I wouldn't worry for now